.github/workflows/ci.yml

```diff
@@ -0,0 +1,459 @@
+          # We cannot simply get rid of the 'tests' folders because behat requires vendor/silverstripe/[framework|cms]/tests/behat/serve-bootstrap.php
+          # Recipes are excluded because the unit tests they run are in the required modules, and there's an assumption they'll require a compatible minor branch of the module
+          if ! [[ "${{ github.repository }}" =~ /(recipe|silverstripe-installer) ]]; then
+            echo "Repositiory is a not a recipe, removing unecessary vendor silverstripe unit tests"
+            # Make an exception for 'ExtendTest.php' which is a misnamed TestOnly non-test class
+            php -r 'foreach (["silverstripe", "cwp", "symbiote", "dnadesign", "tractorcow", "bringyourownideas", "colymba"] as $v) { $d = "vendor/$v"; if (!is_dir($d)) continue; foreach (explode("\n", shell_exec("find $d | grep \/tests\/ | grep [a-zA-Z0-9]Test.php")) as $f) { if (preg_match("#ExtendTest\.php#", $f)) continue; if (is_file($f)) unlink($f); } }'
+            # Also remove a few other file that don't match the *Test.php convention and extends SapphireTest
+            if [ -f vendor/silverstripe/assets/tests/php/FilenameParsing/FileIDHelperTester.php ]; then
+              rm vendor/silverstripe/assets/tests/php/FilenameParsing/FileIDHelperTester.php
+            fi
+            if [ -f vendor/silverstripe/framework/tests/php/Forms/NullableFieldTests.php ]; then
+              rm vendor/silverstripe/framework/tests/php/Forms/NullableFieldTests.php
+            fi
+            if [ -f vendor/silverstripe/graphql/tests/Middleware/MiddlewareProcessTestBase.php ]; then
+              rm vendor/silverstripe/graphql/tests/Middleware/MiddlewareProcessTestBase.php
+            fi
+            # Rebuild composer classloader
+            composer dumpautoload -o
+          fi
+
+      - name: Final preparation
+        run: |
+          # Add .env file and create artifacts directory
+          if [[ "${{ matrix.db }}" =~ mysql ]]; then
+            if [ "${{ matrix.db }}" == "mysql57pdo" ]; then
+              cat << EOF > .env
+          SS_DATABASE_CLASS="MySQLPDODatabase"
+          SS_DATABASE_PORT="3357"
+          EOF
+            else
+              cat << EOF > .env
+          SS_DATABASE_CLASS="MySQLDatabase"
+          SS_DATABASE_PORT="${{ matrix.db == 'mysql57' && '3357' || '3380' }}"
+          EOF
+            fi
+            cat << EOF >> .env
+          SS_DATABASE_SERVER="127.0.0.1"
+          SS_DATABASE_USERNAME="root"
+          SS_DATABASE_PASSWORD="root"
+          EOF
+          else
+            cat << EOF > .env
+          SS_DATABASE_CLASS="PostgreSQLDatabase"
+          SS_DATABASE_SERVER="localhost"
+          SS_DATABASE_PORT="5432"
+          SS_DATABASE_USERNAME="postgres"
+          SS_DATABASE_PASSWORD="postgres"
+          EOF
+          fi
+          cat << EOF >> .env
+          SS_ENVIRONMENT_TYPE="dev"
```

Comment:
Is there ever a scenario in which we should be testing in 'test' or 'live' mode? Or is that something that can be (and is being) handled in unit and behat tests already?

Reply:
I don't think there is?

Reply:
I guess the second part of that is the important part:
> is that something that can be (and is being) handled in unit and behat tests already?

If we're not already doing _some_ tests in live/test mode, we absolutely 100% definitely should be, so if that's not something that our existing tests handle directly, it might be something we should handle at the CI config level.

Reply:
Updating unit/behat tests to support non-dev mode seems way outside the scope of this PR

I don't want to add support for features we haven't agreed are even required

Reply:
I really just wanted an answer to the one question - it's okay if you don't know the answer, that itself is a valid answer to the question. I wasn't requesting any specific changes to this PR. I'll just create a new issue and we can hopefully get an answer to my question over there :p
https://github.com/silverstripe/gha-ci/issues/5